pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -232,6 +250,188 @@ func getMachineDefaulterOperation(platformStatus *osconfigv1.PlatformStatus) mac
 	}
 }
 
+// NewValidatingWebhookConfiguration creates a validation webhook configuration with configured Machine and MachineSet webhooks
+func NewValidatingWebhookConfiguration() *admissionregistrationv1.ValidatingWebhookConfiguration {
+	validatingWebhook := &admissionregistrationv1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookConfigurationName,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/inject-cabundle": "true",
+			},
+		},
+		Webhooks: []admissionregistrationv1.ValidatingWebhook{
+			MachineValidatingWebhook(),
+			MachineSetValidatingWebhook(),
+		},
+	}
+
+	// Setting group version is required for testEnv to create unstructured objects, as the new structure sets it on empty strings
+	// Usual way to populate those values, is to create the resource in the cluster first, which we can't yet do.
+	validatingWebhook.SetGroupVersionKind(admissionregistrationv1.SchemeGroupVersion.WithKind("ValidatingWebhookConfiguration"))
+	return validatingWebhook
+}
+
+// MachineValidatingWebhook returns validating webhooks for machine to populate the configuration
+func MachineValidatingWebhook() admissionregistrationv1.ValidatingWebhook {
+	machineServiceReference := admissionregistrationv1.ServiceReference{
```

Comment:
why `machineServiceReference`? shouldn't this just be `serviceReference` or `webhookServiceReference`?